tests/operator/suite.go

```diff
@@ -428,3 +438,78 @@ func testMultipleSameOperators(check *checksdb.Check, env *provider.TestEnvironm
 
 	check.SetResult(compliantObjects, nonCompliantObjects)
 }
+
+//nolint:funlen
+func testOperatorCatalogSourceBundleCount(check *checksdb.Check, env *provider.TestEnvironment) {
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+
+	const (
+		bundleCountLimit = 1000
+	)
+
+	bundleCountLimitStr := strconv.Itoa(bundleCountLimit)
+
+	// Loop through all labeled operators and check if they have more than 1000 referenced images.
+	var catalogsAlreadyReported []string
+	for _, op := range env.Operators {
+		catalogSourceCheckComplete := false
+		check.LogInfo("Checking bundle count for operator %q", op.Csv.Name)
+
+		cvi := GetCsvVersion(op.Csv.Name)
+
+		// Search through packagemanifests to match the name of the CSV.
+		for _, pm := range env.AllPackageManifests {
+			// Skip all package manifests that do not match the name of the CSV.
+			if pm.Name != cvi.Name {
```

Comment:
This filter won't work always, as the packagemanifest name and the csv "name" are not required (unfortunately) to match. I've been doing some tests with the postgresql operator. See the package (csv) names for the "postgresql" packagemanifest:
```
$ oc get packagemanifests postgresql -o json | jq '.status.channels[0].entries'
[
  {
    "name": "postgresoperator.v5.7.1",
    "version": "5.7.1"
  },
  {
    "name": "postgresoperator.v5.7.0",
    "version": "5.7.0"
  },
  ...
  
$ oc get csv -n test-postgresql-0 
NAME                      DISPLAY                           VERSION   REPLACES                  PHASE
postgresoperator.v5.7.0   Crunchy Postgres for Kubernetes   5.7.0     postgresoperator.v5.6.1   Succeeded
```
I believe the "catalogsource.SkipPMBasedOnChannel()" at line 469 should work for this purpose.

Reply:
Interesting.  Yeah that check won't be very useful if the names aren't required to match.  I'll rework that.